File: atcoder/chokudai_S002_h.cpp
// https://atcoder.jp/contests/chokudai_S002/tasks/chokudai_S002_h
#include<iostream>
#include<algorithm>
//#include <numeric>
//#include<string>
//#include<vector>
//#include<map>
//#include<tuple>
//#include<queue>
//#include<regex>

// #include <bits/stdc++.h>
using namespace std;

int main() {
	long long N, A, B, ans;
	cin >> N;

	for (long long i = 0; i < N; i++)
	{
		cin >> A >> B;

		if (A == B)
		{
			ans = -1;
		}
		else
		{
			ans = abs(A - B);
			//int mx = max(A, B);
			//for (long long j = mx; j > 0; j--)
			//{
			//	if (A % j == B % j)
			//	{
			//		ans = j;
			//		break;
			//	}
			//}
		}

		cout << ans << endl;
	}
}

File: atcoder/chokudai_S002_h.py
# https://atcoder.jp/contests/chokudai_S002/tasks/chokudai_S002_h

N = int(input())
# A, B = [], []
for i in range(N):
    a, b = map(int, input().split())
    # A.append(_a)
    # B.append(_b)
    if a == b:
        ans = -1
    else:
        tmp = max(a, b)
        for i in range(tmp, 0, -1):
            if a % i == b % i:
                ans = i
                break
    print(ans)
